tests/test_layers.py

```diff
@@ -120,6 +120,7 @@ def test_adaptive_embedding(vocab_size, cutoffs, embed_size, units, div_val):
                           [1000, None, 1.0]])
 @pytest.mark.parametrize('embed_size', [128])
 @pytest.mark.parametrize('in_units', [16])
+# TODO This test even passes without sharing the parameters. It needs to be improved.
```

Comment:
Why?

Reply:
If that's the case, we should revise the test. (May be in a later PR).

Reply:
Yes, we should revise the test in a later PR. I just noticed that the test passed prior even when I disabled the parameter sharing or even when the parameter sharing is handled wrongly (such as using invalid regex in `collect_params`)

Reply:
Need to investigate the test after this gets merged. I'm also having issues about parameter sharing in the layout PR so need to wait for this one.